lisa/sut_orchestrator/azure/features.py

```diff
@@ -689,22 +690,38 @@ def setup_rdma(self) -> None:
         if len(devices) > 1:
             # upgrade waagent to latest version to resolve
             # multiple ib devices not getting ip address issue
-            waagent.upgrade_from_source()
+            waagent.upgrade_from_source("v2.9.0.4")
```

Comment:
What's this version? Is it a beta version or released version? Is this step to upgrade or downgrade? Hardcoded version is going to old. Is there a better way to handle it?

Reply:
This is a officially released version but not the latest one. And this step is still upgrading.

The latest walinuxagent source does not work when testing Ubuntu 22.04 marketplace image (canonical 0001-com-ubuntu-server-jammy 22_04-lts 22.04.202307010). No IP address was assigned to ib0. But it works when switching back to the older v2.9.0.4. 

Needs further digging on this one.

Reply:
The same error can be reproed on both Ubuntu 2204 and Mariner v2 marketplace images when using the latest WALinuxAgent source version. But this error won't show up if switching to the prior 2.9.0.4 version.

Was able to discover error logs in waagent.log related to setting up RDMA device and thus based on current investigation, this error seems like a regression in the latest WALinuxAgent source code. Have filed a [bug](https://github.com/Azure/WALinuxAgent/issues/3025) in the WALinuxAgent repository for the WALinuxAgent team to triage. 

For now, the workaround solution is to temporarily upgrade WALinuxAgent to the prior v2.9.0.4 until a conclusion is provided by the WALinuxAgent team regarding this matter.

Reply:
Ok, please keep tracking the bug.